.github/workflows/vast.yaml

```
@@ -140,110 +142,107 @@ jobs:
+          # Run all if this is a release.
+          if [[ "$GITHUB_EVENT_NAME" == "release" ]]; then
+            echo "::notice Enabling release jobs"
+            run_all_checks
+            # Publish pyvast to Pypi.
+            echo "run-python-package=true" >> $GITHUB_OUTPUT
             exit 0
           fi
-          run_python=false
-          if ! is_unchanged python/; then
-            run_python=true
-          fi
-          echo "run-python=${run_python}" >> $GITHUB_OUTPUT
-          run_vast_io=${run_python}
-          if ! is_unchanged web/; then
-            run_vast_io=true
-          fi
-          echo "run-vast-io=${run_vast_io}" >> $GITHUB_OUTPUT
-          run_changelog=${run_vast_io}
-          if ! is_unchanged CHANGELOG.md changelog/; then
-            run_changelog=true
+          # Run all checks if this is a push to master or a tag.
+          if [[ "$GITHUB_EVENT_NAME" == "push" ]]; then
+            echo "::notice Enabling push jobs"
+            run_all_checks
+            exit 0
           fi
-          echo "run-changelog=${run_changelog}" >> $GITHUB_OUTPUT
-          run_vast=false
-          if ! is_unchanged cmake/ CMakeLists.txt libvast/ libvast_test/ scripts/ schema/ tools/ vast/ vast.yaml.example version.json; then
-            run_vast=true
+          # Run only what is requested for a workflow dispatch.
+          if [[ "$GITHUB_EVENT_NAME" == "workflow_dispatch" ]]; then
+            echo "::notice Enabling workflow dispatch jobs"
+            if [[ "${{ inputs.upload-static-binary-to-github }}" == "true" ]]; then
+              echo "run-vast-nix=true" >> $GITHUB_OUTPUT
+            fi
+            if [[ "${{ inputs.upload-docker-image-to-ecr }}" == "true" ]]; then
+              echo "run-docker-vast=true" >> $GITHUB_OUTPUT
+            fi
+            exit 0
           fi
-          run_vast_plugins=${run_vast}
-          if ! is_unchanged plugins/; then
-            run_vast_plugins=true
+          # From here on we should be in a pull request.
+          if [[ "$GITHUB_EVENT_NAME" != "pull_request" ]]; then
+            echo "::error Unexpected GitHub Event: $GITHUB_EVENT_NAME"
+            exit 1
           fi
-          if ! is_unchanged contrib/vast-plugins/; then
-            run_vast_plugins=true
+          echo "::notice Enabling pull request jobs"
+          echo "::notice sourcing configure helpers"
+          source ./.github/workflows/configure_helpers.bash
+          # Run all checks if this file changed.
+          if is_changed .github/workflows/vast.yaml; then
+            run_all_checks
+            exit 0
           fi
-          echo "run-vast-plugins=${run_vast_plugins}" >> $GITHUB_OUTPUT
-          # Build vast if no artifact is present in google store
+          run_if_changed python "python/"
+          run_if_changed vast-io "web/" "python/"
+          run_if_changed changelog "CHANGELOG.md" "changelog/" "web/" "python/"
```

```suggestion
          run_if_changed changelog "changelog/" "web/" "python/"
```

That file doesnt exist anymore, actually.